source/slang/slang-ir-inst-defs.h

```diff
@@ -166,7 +166,14 @@ INST(Nop, nop, 0, 0)
 INST(StructType, struct, 0, PARENT)
 INST(InterfaceType, interface, 0, PARENT)
 
-INST_RANGE(Type, VoidType, InterfaceType)
+// A TypeType-typed IRValue represents a IRType.
+// It is used to represent a type parameter/argument in a generics.
+INST(TypeType, type_t, 0, 0)
+
+// An `IRWitnessTable` has type `WitnessTableType`.
+INST(WitnessTableType, witness_table_t, 0, 0)
+
+INST_RANGE(Type, VoidType, WitnessTableType)
```

Comment:
Similarly, but less important: we should consider whether witness tables are classified by a `WitnessTableType` or by a `WitnessTableKind`. For the most part they are type-level entities in practice, so I suppose that `WitnessTableKind` might be more accurate.

The same basic choice space arises as for `TypeType` and `TypeKind`, where we need to decide whether to use a single classifier for all cases (both generic and value-level parameters), or to distinguish the use cases as different classifiers.